apps/opik-backend/src/main/java/com/comet/opik/infrastructure/bi/UsageReportDAO.java

```diff
@@ -0,0 +1,80 @@
+package com.comet.opik.infrastructure.bi;
+
+import com.google.inject.ImplementedBy;
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.core.statement.UnableToExecuteStatementException;
+
+import java.sql.SQLIntegrityConstraintViolationException;
+import java.util.Optional;
+
+@ImplementedBy(UsageReportDAOImpl.class)
+interface UsageReportDAO {
+
+    Optional<String> getAnonymousId();
+
+    void saveAnonymousId(@NonNull String id);
+
+    boolean isEventReported(@NonNull String eventType);
+
+    void addEvent(@NonNull String eventType);
+
+    void markEventAsReported(@NonNull String eventType);
+}
+
+@Slf4j
+@RequiredArgsConstructor(onConstructor_ = @Inject)
+@Singleton
+class UsageReportDAOImpl implements UsageReportDAO {
+
+    private final Jdbi jdbi;
+
+    public Optional<String> getAnonymousId() {
+        return jdbi.inTransaction(handle -> handle.createQuery("SELECT value FROM metadata WHERE `key` = :key")
```

Comment:
The implementation of this DAO follows the declarative API of JDBI3, whereas everything else in the service follows the fluent API. You should be consistent here.